File: CTask/src/org/ShapeSHifter420/Filters/TechFilter.java
package org.ShapeSHifter420.Filters;

import org.ShapeSHifter420.Pushes.*;

public class TechFilter implements IFilter {
    public boolean isAllow(IPush push){
        return push instanceof ITechPush;
    };
    public boolean test(IPush push, ISystem sys){
        ITechPush agepush = (ITechPush) push;
        return  (agepush.GetOSVersion() >= sys.GetOSVersion());
    }
}

File: CKotlin/Filters/TechFilter.kt
package Filters

import Pushes.IPush
import Pushes.ISystemData
import Pushes.ITechPush

class TechFilter : IFilter {
    override fun allow(push: IPush): Boolean = push is ITechPush

    override fun filtration(push: IPush,sysData: ISystemData): Boolean {
        val techPush = push as ITechPush
        return sysData.osVersion <= techPush.osVersion
    }
}